core/vote/vote_manager.go

```diff
@@ -34,7 +32,38 @@ type VoteManager struct {
 
 	engine consensus.PoSA
 
-	getJustifiedHeader getJustifiedHeaderFunc
+	voteDataBuffer *voteDataBuffer
+}
+
+type voteDataBuffer struct {
+	voteDataMap map[uint64]*types.VoteData
+	cache       []uint64
+}
+
+func newVoteDataBuffer() *voteDataBuffer {
+	return &voteDataBuffer{
+		voteDataMap: make(map[uint64]*types.VoteData, maxSizeOfRecentEntry),
```

Comment:
Can we use the lru.cache, then we can manage the size easier?

Reply:
Refer diffLayerCache             *lru.Cache   // Cache for the diffLayers

Reply:
Done